autogpt/agent/agent.py

```diff
@@ -119,111 +119,25 @@ def start_interaction_loop(self):
                     cfg.fast_token_limit,
                 )  # TODO: This hardcodes the model to use GPT3.5. Make this an argument
 
-            assistant_reply_json = fix_json_using_multiple_techniques(assistant_reply)
-            for plugin in cfg.plugins:
-                if not plugin.can_handle_post_planning():
-                    continue
-                assistant_reply_json = plugin.post_planning(assistant_reply_json)
-
-            # Print Assistant thoughts
+            assistant_reply_json = self._get_json_assistant_reply(assistant_reply, cfg)
             if assistant_reply_json != {}:
-                validate_json(assistant_reply_json, LLM_DEFAULT_RESPONSE_FORMAT)
-                # Get command name and arguments
                 try:
-                    print_assistant_thoughts(
-                        self.ai_name, assistant_reply_json, cfg.speak_mode
-                    )
+                    # Get command name and arguments
                     command_name, arguments = get_command(assistant_reply_json)
-                    if cfg.speak_mode:
-                        say_text(f"I want to execute {command_name}")
-
                     arguments = self._resolve_pathlike_command_args(arguments)
-
                 except Exception as e:
                     logger.error("Error: \n", str(e))
-            self.log_cycle_handler.log_cycle(
-                self.config.ai_name,
-                self.created_at,
-                self.cycle_count,
-                assistant_reply_json,
-                NEXT_ACTION_FILE_NAME,
-            )
-
-            logger.typewriter_log(
-                "NEXT ACTION: ",
-                Fore.CYAN,
-                f"COMMAND = {Fore.CYAN}{command_name}{Style.RESET_ALL}  "
-                f"ARGUMENTS = {Fore.CYAN}{arguments}{Style.RESET_ALL}",
-            )
+                else:
+                    self._print_assistant_thoughts(
+                        arguments, assistant_reply_json, cfg, command_name
+                    )
```

Comment:
This part used to be in same try-except block with get_command.
Getting command and printing thoughts seems to be separate behavior so it looks they could be split.
Also this looked like it should only be executed if there isn't any error, so I moved it to else block.